horace_core/sqw/@Experiment/private/build_from_old_headers_.m

```diff
@@ -88,5 +159,82 @@
         hdr.samples = hdr.sample;
         hdr = rmfield(hdr,'sample');
     end    
+    
+    % Repack the header struct into the cell array
     headers{i} = hdr;
+end % numel(headers)
+
+end % function normalise_old_headers
+
+%-------------------------------------------------------------------------
+function instr = make_instrument_from_struct(instr_info)
+%
+% Temporary location for this instrument factory which should really go in
+% Herbert instrument classes location
+%
+% Given info on e.g. moderator, aperture, chopper, constructs a specific
+% instrument from the pre-defined instruments
+% 
+% Input
+% -----
+%   >> instr_info: struct with fields for moderator, aperture, chopper
+%                  but not otherwise well defined (TODO)
+%
+% Output
+% ------
+%   >> instr: the constructed instrument, if successful
+%
+% Errors
+% ------
+% If the struct does not sufficiently define one of the predefined
+% instruments, an error will be thrown
+
+if ~isfield(instr_info, 'moderator') || ~isfield(instr_info, 'aperture')
+    error('HORACE:Experiment:invalid_argument',...
+        'unknown structure tried to be recovered as instrument [1]');
+end
+
+if isfield(instr_info,'fermi_chopper') 
+    instr = IX_inst_DGfermi(instr_info.moderator, ...
+        instr_info.aperture,  ...
+        instr_info.fermi_chopper);
+elseif isfield(instr_info,'disk_chopper') 
+    instr = IX_inst_DGdisk(instr_info.moderator, ...
+        instr_info.aperture,  ...
+        instr_info.disk_chopper);
+else
+    error('HORACE:Experiment:invalid_argument',...
+        'unknown structure tried to be recovered as instrument [2]');
 end
+
+end % function make_instrument_from_struct
+
+%-------------------------------------------------------------------------
+function sample = make_sample_from_struct(~) % when used, change ~ to sample_info)
+%
+% Temporary location for this sample factory which should really go in
+% Herbert instrument classes location
+%
+% Given info on e.g. moderator, aperture, chopper, constructs a specific
+% instrument from the pre-defined instruments
+% 
+% Input
+% -----
+%   >> sample_info: struct with fields for sample things (TODO)
+%
+% Output
+% ------
+%   >> sample: the constructed sample, currently not implemented, so fails
+%
+% Errors
+% ------
+% If the struct does not sufficiently define a sample (which it won't, not
+% defined yet), an error will be thrown
+
+sample = IX_null_sample(); % suppress error, should be removed when proper 
```

Comment:
Is proper functionality to be added?

Reply:
This is an implementation gone off half-cocked. Thanks, will fix